api/envoy/extensions/filters/common/dependency/v3/dependency.proto

```diff
@@ -42,3 +42,14 @@ message FilterDependencies {
   // A list of dependencies provided on the encode path.
   repeated Dependency encode_provided = 4;
 }
+
+// Matching requirements for a filter. For a match tree to be used with a filter,
+// the match requirements must be satisfied.
```

Comment:
Sorry I meant actual end user docs. :)

Can you add some example somewhere that uses this with the rest of your match docs?

/wait

Reply:
Sure, added some. The big user facing thing that's missing at the moment is a way to expose the restrictions of each filter in the docs, but I was hoping to piggy back on the work in https://github.com/envoyproxy/envoy/issues/14470 to have these matching requirements be specified in the same manifest as the filter dependencies, allowing us to render it as part of per filter docs.

Reply:
I think I'm still missing something. Where is this message used? Can you add a fully formed config example somewhere? Is the idea that these are hard coded and returned by the filter itself? If so I think I understand, but maybe make that slightly more clear that it's provided by the filter and not by the user?

Reply:
I think more examples aren't necessarily needed, I would just make it more clear to the reader that this is provided by the filter rather than the user.